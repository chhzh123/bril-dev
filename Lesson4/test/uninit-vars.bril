# ARGS: uninit_var

@main(cond: bool) {
  br cond .left .right;
.left:
  a: int = const 1;
  b: int = const 5;
  jmp .end;
.right:
  a: int = const 2;
  c: int = const 10;
  jmp .end;
.end:
  d: int = sub a c;
  e: int = add b c;
  print d;
}