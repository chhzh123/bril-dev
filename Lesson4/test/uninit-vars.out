b1:
  in:  d, c, cond, b, a
  out: d, c, cond, b, a
left:
  in:  cond, d, c, b, a
  out: cond, d, c
right:
  in:  cond, d, c, b, a
  out: cond, d, b
end:
  in:  cond, d, c, b
  out: cond, c, b
